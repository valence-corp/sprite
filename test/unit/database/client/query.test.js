"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const testClient_js_1 = require("./testClient.js");
const database_js_1 = require("../../../../src/endpoints/database.js");
const variables_js_1 = require("../../../variables.js");
describe('SpriteDatabase.query()', () => {
    it(`should make a properly formatted POST request to ${database_js_1.endpoints.query}/${variables_js_1.variables.databaseName}`, async () => {
        jest.spyOn(global, 'fetch').mockResolvedValueOnce({
            status: 200,
            json: async () => variables_js_1.variables.jsonResponse,
        });
        await testClient_js_1.client.query('gremlin', variables_js_1.variables.nonEmptyString);
        expect(global.fetch).toHaveBeenCalledWith(`${variables_js_1.variables.address}${database_js_1.endpoints.query}/${variables_js_1.variables.databaseName}`, {
            method: 'POST',
            headers: {
                'Authorization': `Basic ${variables_js_1.testAuth}`,
                'Content-Type': 'application/json',
            },
            body: JSON.stringify({
                language: 'gremlin',
                command: variables_js_1.variables.nonEmptyString,
            }),
        });
    });
    it('should handle a 200 response by returning the data from the response', async () => {
        jest.spyOn(global, 'fetch').mockResolvedValueOnce({
            status: 200,
            json: async () => variables_js_1.variables.jsonResponse,
        });
        const result = await testClient_js_1.client.query('gremlin', variables_js_1.variables.nonEmptyString);
        expect(result).toMatchSnapshot();
    });
    it('should handle a 400 response by throwing an error', async () => {
        jest.spyOn(global, 'fetch').mockResolvedValueOnce({
            status: 400,
        });
        await expect(testClient_js_1.client.query('sql', variables_js_1.variables.nonEmptyString)).rejects.toMatchSnapshot();
    });
    it('should handle a 500 response by throwing an error', async () => {
        jest.spyOn(global, 'fetch').mockResolvedValueOnce({
            status: 500,
        });
        await expect(testClient_js_1.client.query('gremlin', variables_js_1.variables.nonEmptyString)).rejects.toMatchSnapshot();
    });
    it('should handle an unexpected status code by throwing an error', async () => {
        jest.spyOn(global, 'fetch').mockResolvedValueOnce({
            status: 418,
            statusText: variables_js_1.variables.nonEmptyString,
        });
        await expect(testClient_js_1.client.query('gremlin', variables_js_1.variables.nonEmptyString)).rejects.toMatchSnapshot();
    });
});
//# sourceMappingURL=query.test.js.map