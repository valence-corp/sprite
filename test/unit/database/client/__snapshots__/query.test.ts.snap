// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpriteDatabase.query() should handle a 200 response by returning the data from the response 1`] = `
{
  "result": "test",
}
`;

exports[`SpriteDatabase.query() should handle a 400 response by throwing an error 1`] = `[Error: Invalid language or query. Status: 400]`;

exports[`SpriteDatabase.query() should handle a 500 response by throwing an error 1`] = `[TypeError: response.json is not a function]`;

exports[`SpriteDatabase.query() should handle an unexpected status code by throwing an error 1`] = `[Error: Unknown error. Status: 418, StatusText: non-empty string]`;
